SG14/slot_map.h

```diff
@@ -132,24 +128,20 @@ class slot_map
     // If the check fails it is undefined behavior.
     // O(1) time and space complexity.
     //
-    constexpr reference operator[](const key_type& key)              { return *find_unchecked(key); }
-    constexpr const_reference operator[](const key_type& key) const  { return *find_unchecked(key); }
+    constexpr reference operator[](const key_type& key) {
+        return const_cast<reference>(static_cast<const slot_map*>(this)->operator [](key));
+    }
+    constexpr const_reference operator[](const key_type& key) const {
+        return *find_unchecked(key);
+    }
 
     // The find() functions have generation counter checking.
     // If the check fails, the result of end() is returned.
     // O(1) time and space complexity.
     //
     constexpr iterator find(const key_type& key) {
-        auto slot_index = get_index(key);
-        if (slot_index >= slots_.size()) {
-            return end();
-        }
-        auto slot_iter = std::next(slots_.begin(), slot_index);
-        if (get_generation(*slot_iter) != get_generation(key)) {
-            return end();
-        }
-        auto value_iter = std::next(values_.begin(), get_index(*slot_iter));
-        return value_iter;
+        const_iterator it = static_cast<const slot_map*>(this)->find(key);
+        return values_.erase(it, it);
```

Comment:
This diff seems wrong.
If this diff is indeed wrong, _and_ if the unit-tests don't currently catch it, would you mind opening a new PR with a single new unit test that would have caught this?

Reply:
What do you mean by "the diff is wrong"? Erase is used to convert from const_iterator to iterator only here. It is the recommended technique to "const_cast" an iterator.

Reply:
Oh, I see, you're erasing a range of length _zero_ (not length one). Still, rather than require the container to provide a "`const_iterator_cast`" spelled `erase`, we should either not require `const_iterator_cast` at all (my preference), or else require it to be provided via a specific customization point.